### PR TITLE
feat: Add typings

### PR DESCRIPTION
**Pull request by aren55555**
Feature request: https://github.com/Finch-API/react-finch-connect/issues/4

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    payrollProvider?: string;
+    sandbox?: boolean;
+    onSuccess?: (e: SuccessEvent) => void;
+    onError?: (e: ErrorEvent) => void;
+    onClose?: () => void;
+    zIndex?: bigint | string;
+  }
+
+  export function useFinchConnect(opts: ConnectOptions): { open: () => void };
+}
```
Comment (skilbjo) [inline src/index.d.ts:24]: seems a little unusual to me to include this is `src/`

what i would expect:

typescript types are declared inline in your source code. ie `SuccessEvent` is defined in `src/example/App.tsx` (for example), and then when you run `npm pack` (during a release), the compiler handles the breaking out of the `dist/foo.d.ts` and `dist/foo.js` files. the contents of `dist/` are then what get sent to npm / github package registry. 

however two caveats to my comment:
- that's what i'm used to on server side typescript, i'm unfamiliar w client side ts
- this approach will work as well, and nothing seems wrong with it

File: rollup.config.js [hunk 2]
```diff
@@ -1,32 +1,45 @@
...
-    external(),
-    url({ exclude: ['**/*.svg'] }),
-    babel({
-      exclude: 'node_modules/**',
-    }),
-    resolve(),
-    commonjs(),
-  ],
-};
+export default [
```
Comment (skilbjo) [inline rollup.config.js:10]: i am unfamiliar with this build tool so can't review appropriately here

File: example/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      "dom.iterable",
+      "esnext"
+    ],
+    "allowJs": true,
+    "skipLibCheck": true,
+    "esModuleInterop": true,
+    "allowSyntheticDefaultImports": true,
+    "strict": true,
+    "forceConsistentCasingInFileNames": true,
+    "module": "esnext",
```
Comment (skilbjo) [inline example/tsconfig.json:15]: should this be `commonjs` ?

**Review by skilbjo — APPROVED** [review]

File: src/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+declare module 'react-finch-connect' {
+  export interface SuccessEvent {
```
Comment (cathy-chi) [inline src/index.d.ts:2]: Using types here would be better than interface.

**Pull request merged**